### PR TITLE
Add bot_token argument

### LINKED ISSUE #123: Allow specifying custom file names for bot sessions
Currently, the session file name is named after the bot id, not really nice in case we wan't to tell multiple sessions apart. Having the possibility to specify a custom file name (just like user auths do) would be nice.

### PR DESCRIPTION
**Pull request by bakatrouble**
Closes #123.
Old behavior (bot token as session name) is still remaining with warning, but should be removed in future

### CODE REVIEW

File: pyrogram/client/ext/base_client.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ class StopTransmission(StopIteration):
     }
 
     def __init__(self):
-        self.bot_token = None
+        self.is_bot = False
```
Comment (delivrance) [inline pyrogram/client/ext/base_client.py]: Better set this to `None` because... (read next review)

File: pyrogram/client/client.py [hunk 2]
```diff
@@ -284,11 +295,12 @@ def start(self):
                 if self.bot_token is None:
                     self.authorize_user()
```
Comment (delivrance) [inline pyrogram/client/client.py]: ...because you are relying on `self.is_bot = False` here. Better be explicit and set it to False here instead.

**Comment by delivrance** [comment]
Beside the small review, I'd change nothing else. Nice job

**Comment by bakatrouble** [comment]
> Beside the small review, I'd change nothing else. Nice job

Done.

**Pull request merged**